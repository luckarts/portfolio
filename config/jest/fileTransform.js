const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};