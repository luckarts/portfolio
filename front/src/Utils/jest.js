import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
