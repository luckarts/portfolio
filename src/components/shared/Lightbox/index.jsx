import React from 'react';
import Lightbox from 'react-image-lightbox';
import 'react-image-lightbox/style.css';

import PropTypes from 'prop-types';

const propTypes = {
	lightboxIsOpen: PropTypes.bool.isRequired,
	photoIndex: PropTypes.number.isRequired,
	closeLightbox: PropTypes.func.isRequired,
	getIndexPhoto: PropTypes.func.isRequired,
};

const LightboxGalerie = ({ lightboxIsOpen, photoIndex, closeLightbox, getIndexPhoto, photos }) => {
	return (
		<div>
			{lightboxIsOpen === true && photos[photoIndex] ? (
				<Lightbox
					mainSrc={'/img/' + photos[photoIndex].image}
					nextSrc="1"
					prevSrc="0"
					onCloseRequest={() => closeLightbox(false)}
					onMovePrevRequest={() => getIndexPhoto((photoIndex + photos.length - 1) % photos.length)}
					onMoveNextRequest={() => getIndexPhoto((photoIndex + 1) % photos.length)}
				/>
			) : (
				<div />
			)}
		</div>
	);
};

LightboxGalerie.propTypes = propTypes;

export default LightboxGalerie;
