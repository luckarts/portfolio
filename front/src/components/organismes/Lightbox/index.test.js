import React from "react";
import { shallow, mount } from "enzyme";
import LightboxComponent from ".";
import { checkProps } from "Utils/jest";
import Lightbox from "react-image-lightbox";
let wrapper, props, expectedProps, id, photo, closeLightbox;
const photos = [
  {
    src: "/img/copper_study.jpg",
    width: 4,
    height: 3
  }
];
props = {
  lightboxIsOpen: true,
  photoIndex: 0,
  closeLightbox: jest.fn(),
  getIndexPhoto: jest.fn()
};

it("Should render.", () => {
  expect(mount(<LightboxComponent {...props} />)).toMatchSnapshot();
});

describe("Galerie Component", () => {
  beforeEach(() => {
    wrapper = shallow(<LightboxComponent {...props} />);
  });
  it(" Should call closeLightbox", () => {
    wrapper.find(Lightbox).prop("onCloseRequest")(false);
    expect(props.closeLightbox).toHaveBeenCalledWith(false);
  });
  it(" Should call onMoveNextRequest", () => {
    wrapper.find(Lightbox).prop("onMoveNextRequest")(props.photoIndex);
    expect(props.getIndexPhoto).toHaveBeenCalledWith(1);
  });
  it(" Should call onMoveNextRequest", () => {
    wrapper.find(Lightbox).prop("onMoveNextRequest")(props.photoIndex);
    expect(props.getIndexPhoto).toHaveBeenCalledWith(1);
  });
  it(" Should have props mainSrc ", () => {
    wrapper.find(Lightbox).prop("mainSrc");
    expect(wrapper.find(Lightbox).prop("mainSrc")).toEqual(
      "/img/copper_study.jpg"
    );
  });
});
