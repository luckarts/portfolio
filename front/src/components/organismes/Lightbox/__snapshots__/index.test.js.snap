// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render. 1`] = `ReactWrapper {}`;

exports[`renders correctly 1`] = `[Function]`;
